packages/schematics/update/update/index.ts

```diff
@@ -826,6 +827,13 @@ function _formatVersion(version: string | undefined) {
   return version;
 }
 
+// Returns whether or not the given package spec (the value string in a
+// `package.json` dependency) is hosted in the NPM registry.
+function isPkgFromRegistry(name: string, spec: string): boolean {
+  const result = npa.resolve(name, spec);
```

Comment:
This can throw if the name or spec is invalid.  In those cases, this should probably return false since update will fail when it attempts to analyze the package.

Reply:
I was thinking that in the case of an invalid package name, we should just abort and let the update process crash. If this throws, then `package.json` must be malformed or corrupted and would not be usable by NPM either. Is that a case where we would want to the update process to suppress the error and skip that package?

Reply:
Barring catastrophic errors such as out of memory exceptions, the CLI itself should ideally never crash.
However, since the update logic is currently wrapped in a schematic, the only way to report an error is to throw an exception so that should work.  The exception messages should probably be augmented and rethrown with additional information/instructions regarding the failure so that they are actionable by the user though.

Reply:
"crash" was the wrong word for me to use, but yes I agree. Updated to rethrow a more meaningful error message to the user. This does throw directly from the function rather than emit an error from a returned `Observable`. I think this is ok based on the fact that `_getAllDependencies()` also throws and I'm assuming there is a global error handler which will catch and print the message.